### PR TITLE
update DG and UG

### PR DESCRIPTION
**Pull request by yan-soon**
**DG**

- Update Logic Component class diagram and description
- Update delete command sequence diagram (under Logic Component as an example)
- Add implementation details for add command
- Add new Add command sequence diagram

**UG**

- Update add command description to be clearer
- Update add command to include prefixes (/c, /cr, /t) in the code format example

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -43,9 +43,11 @@ title: User Guide
     
 1. Adding a module
 
-    * Allows user to specify module description
+    * Allows user to add any Module of choice into GradPad for tracking purposes.
```
Comment (shaokiat) [inline docs/UserGuide.md]: Perhaps can specify that modules are added into the Current Modules section/column instead of just GradPad

Comment (yan-soon) [reply docs/UserGuide.md]: Okay, updated the description! Thanks for feedback.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -133,6 +133,44 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+Given below is how an add operation behaves at each step of its execution.
+
+Step 1. The user types in a command string corresponding to an add operation.
+
+Step 2. This calls the `execute` method of the `LogicManager` class. The user input is passed in as a string.
+
+Step 3. `Logic.execute()` then calls the `parseCommand` method of the `gradPadParser` class to parse the string input.
+
+Step 4. `gradPadParser.parseCommand()` sees that this is an add command, and so uses the `AddCommandParser`
+class to create a corresponding `EditCommand`.
```
Comment (Silvernitro) [inline docs/DeveloperGuide.md]:
```suggestion
class to create a corresponding `AddCommand`.
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -133,6 +133,44 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+Step 5. In `AddCommandParser`, the string input is first split into tokens, i.e. new module code, new tags, etc.
+
+Step 6. Then, in the same method call, a new `Module` object is created from these tokens. It now stores
```
Comment (Silvernitro) [inline docs/DeveloperGuide.md:160]: I think the method call wasn't specified; is it supposed to be `AddCommandParser.parse()`?

File: docs/diagrams/DeleteSequenceDiagram.puml [hunk 4]
```diff
@@ -3,7 +3,7 @@
```
Comment (Silvernitro) [inline docs/diagrams/DeleteSequenceDiagram.puml:3]: We need to change `deletePerson(1)` in line 51 :grin:

File: docs/diagrams/DeleteSequenceDiagram.puml [hunk 5]
```diff
@@ -36,13 +36,13 @@ activate DeleteCommand
...
+DeleteCommandParser --> GradPadParser : d
 deactivate DeleteCommandParser
 'Hidden arrow to position the destroy marker below the end of the activation bar.
-DeleteCommandParser -[hidden]-> AddressBookParser
+DeleteCommandParser -[hidden]-> GradPadParser
 destroy DeleteCommandParser
 
-AddressBookParser --> LogicManager : d
+GradPadParser --> LogicManager : d
 deactivate AddressBookParser
```
Comment (Silvernitro) [inline docs/diagrams/DeleteSequenceDiagram.puml]:
```suggestion
deactivate GradPadParser
```

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 6]
```diff
@@ -81,7 +81,7 @@ public CommandResult execute(Model model) throws CommandException {
 
     /**
      * Creates and returns a {@code Module} with the details of {@code moduleToEdit}
-     * edited with {@code editModuleDescriptor}.
+     * edited with {@code editModuleFDescriptor}.
```
Comment (Silvernitro) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: typo? hehe

File: docs/diagrams/AddSequenceDiagram.puml [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+activate Module
+
+Module --> AddCommandParser
+deactivate Module
+
+create AddCommand
+AddCommandParser -> AddCommand : AddCommand(Module)
+activate AddCommand
+
+AddCommand --> AddCommandParser : e
```
Comment (Silvernitro) [inline docs/diagrams/AddSequenceDiagram.puml]: Maybe we can rename the return variable as `a` or smth? I only named it `e` in my diagram cos it stands for `EditCommand` :grin:

File: docs/UserGuide.md [hunk 8]
```diff
@@ -105,9 +107,9 @@ Expected outcome:
 
 Adds a new module.
 
-Format: `add {module code} {modular credits}`
+Format: `add /c {module code} /cr {modular credits}`
```
Comment (Silvernitro) [inline docs/UserGuide.md]:
```suggestion
Format: `add c/ {module code} cr/ {modular credits}`
```

**Review by Silvernitro — COMMENTED** [review]
Just a new nits!

File: docs/UserGuide.md [hunk 9]
```diff
@@ -105,9 +107,9 @@ Expected outcome:
...
 
-Example of usage: `add CS2103T 4`
+Example of usage: `add /c CS2103T /cr 4`
```
Comment (shaokiat) [inline docs/UserGuide.md]:
```suggestion
Example of usage: `add c/ CS2103T cr/ 4`
```

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (yan-soon) [reply docs/DeveloperGuide.md]: Updated!

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (yan-soon) [reply docs/DeveloperGuide.md:160]: Updated to include entire method name!

File: docs/diagrams/DeleteSequenceDiagram.puml (hunk 4 above)
Comment (yan-soon) [reply docs/diagrams/DeleteSequenceDiagram.puml:3]: Changed to deleteModule(1)!

File: docs/diagrams/DeleteSequenceDiagram.puml (hunk 5 above)
Comment (yan-soon) [reply docs/diagrams/DeleteSequenceDiagram.puml]: Updated!

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 6 above)
Comment (yan-soon) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: Wah this one rly good find...

File: docs/diagrams/AddSequenceDiagram.puml (hunk 7 above)
Comment (yan-soon) [reply docs/diagrams/AddSequenceDiagram.puml]: Renamed!

File: docs/UserGuide.md (hunk 8 above)
Comment (yan-soon) [reply docs/UserGuide.md]: Changed!

File: docs/UserGuide.md (hunk 9 above)
Comment (yan-soon) [reply docs/UserGuide.md]: Changed!

**Review by shaokiat — APPROVED** [review]
LGTM! 💯

**Review by Silvernitro — APPROVED** [review]

**Comment by mhdsyfq77** [comment]
LGTM bro! good job 😄

**Pull request merged**